Sources/Sentry/SentryScope.m

```diff
@@ -330,6 +364,13 @@ - (void)addAttachment:(SentryAttachment *)attachment
     [serializedData setValue:self.environmentString forKey:@"environment"];
     [serializedData setValue:[self fingerprints] forKey:@"fingerprint"];
 
+    if (self.currentTransaction != nil &&
+        [self.currentTransaction isKindOfClass:[SentryTracer class]])
+        [serializedData setValue:[(SentryTracer *)self.currentTransaction name]
+                          forKey:@"transaction"];
+    else if (self.fallbackTransactionName != nil)
+        [serializedData setValue:self.fallbackTransactionName forKey:@"transaction"];
```

Comment:
question, should we do that for transactions too? @philipphofmann 

on Android, if there's a hard crash, the transaction won't be sent because it's not finished by the user and we don't finish transactions/spans automatically.
Transactions also don't run `beforeSend` so they could not finish it in there too.

@bruno-garcia maybe we should finish the transaction automatically with an error if its bound ot the Scope when there's a hard crash?

Reply:
That's a good question. I assume it's valuable to see the transaction that was ongoing at the time of the crash. We could consider this but it can also be a next step (not part of this PR/release).

It's something to consider across all mobile SDKs even.

Reply:
yes, agree

Reply:
Good point, but I agree let's not fix this in this PR.